### DISCUSSION TITLE
source_type not documented for file source

### DISCUSSION THREAD

**Discussion by tshepang** [vectordotdev/vector#13487] [Q&A] [answered]
Am seeing `"source_type":"file"` in the output of file source, but I see it's not documented here https://vector.dev/docs/reference/configuration/sources/file/#line-fields. I wonder if it is intentional, because it's not documented for the few other sources I checked.

**Comment by neuronull** [comment, marked as answer]
Hi @tshepang !

I think this might be a discrepancy in the documentation.
For example, I see `source_type` documented in the kubernetes source:

https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#output-data

I'll dig into this more but I suspect all sources would have that `source_type` field.

**Reply by neuronull** [reply]
Indeed it appears the `source_type` should be a required field for all sources and that the docs need to be updated.
I'll open a PR to address this shortly.

Thanks for reporting this!

**Reply by neuronull** [reply]
So while updating the docs, it appears that not all of the sources output `source_type` (though they should).

This PR is to update the docs with those sources that currently do : https://github.com/vectordotdev/vector/pull/13541

And this PR was opened to make the `source_type` report at a higher level so that all sources will have it: https://github.com/vectordotdev/vector/issues/13560

Would you mind marking this as the accepted answer if its 👌 ? Thanks!
